templates/_helpers.tpl

```diff
@@ -274,7 +299,12 @@ Sets the injector node selector for pod placement
 {{- define "injector.nodeselector" -}}
   {{- if .Values.injector.nodeSelector }}
       nodeSelector:
-        {{ tpl .Values.injector.nodeSelector . | indent 8 | trim }}
+        {{- $tp := typeOf .Values.injector.nodeSelector }}
+        {{- if eq $tp "string" }}
+          {{ tpl .Values.injector.nodeSelector . | nindent 8 | trim }}
+        {{- else }}
+          {{- toYaml .Values.injector.nodeSelector | nindent 8 }}
+        {{- end }}
```

Comment:
The string case has some extra indenting here when rendered, so we may want to shift the whole block back like this:

```suggestion
      {{- $tp := typeOf .Values.injector.nodeSelector }}
      {{- if eq $tp "string" }}
        {{ tpl .Values.injector.nodeSelector . | nindent 8 | trim }}
      {{- else }}
        {{- toYaml .Values.injector.nodeSelector | nindent 8 }}
      {{- end }}
```

Reply:
Fixed in latest commit.